ios/RNSPeekableView.m

```diff
@@ -0,0 +1,166 @@
+#import "RNSPeekableView.h"
+#import "RNSScreenContainer.h"
+#import <React/RCTUIManagerObserverCoordinator.h>
+#import <React/RCTUIManager.h>
+#import "RNSScreen.h"
+#import <React/RCTUIManagerUtils.h>
+
+
+@implementation RNPreviewViewController {
+  RNSPeekableWrapper *_wrapper;
+}
+
+
+- (instancetype)initWithWrapper:(RNSPeekableWrapper *) wrapper
+{
+  if (self = [super init]) {
+    _wrapper = wrapper;
+  }
+  return self;
+}
+
+- (UIViewController *)previewingContext:(id<UIViewControllerPreviewing>)previewingContext viewControllerForLocation:(CGPoint)location
+{
+  if (_wrapper.onPeek) {
+    _wrapper.onPeek(@{});
+  }
+  return _wrapper.previewController;
+}
+
+- (void)previewingContext:(id<UIViewControllerPreviewing>)  previewingContext
+     commitViewController:(UIViewController *)viewControllerToCommit
+{
+  if (_wrapper.onPop) {
+    _wrapper.onPop(@{});
+  }
+}
+
+- (NSArray<id<UIPreviewActionItem>> *)previewActionItems {
+  return _wrapper.actionsForPreviewing;
+}
+
+- (void)viewDidDisappear:(BOOL)animated {
+  if (_wrapper.onDisappear) {
+    _wrapper.onDisappear(@{});
+  }
+}
+
+@end
+
+
+@implementation RNSPeekableWrapper {
+  RCTUIManager *_uiManager;
+  UIView *_child;
+}
+
+
+- (instancetype)initWithUIManager:(RCTUIManager *)uiManager
+{
+  if (self = [super init]) {
+    _uiManager = uiManager;
+  }
+  _previewController = [[RNPreviewViewController alloc] initWithWrapper:self];
+  
+  return self;
+}
+
+- (void)setOnPop:(RCTDirectEventBlock)onPop
+{
+  _onPop = onPop;
```

Comment:
do we really need all those methods here? seems like sth that should be synthetized automatically